src/utils.rs

```diff
@@ -11,7 +11,23 @@ pub fn _string(raw_ptr: *const c_char) -> String {
 }
 
 pub fn _string_array(raw_ptr: *mut *mut c_char) -> Vec<String> {
-    let mut ret_val: Vec<String> = vec![];
+    _convert_raw_ptr_array(raw_ptr, _string)
+}
+
+pub fn _pathbuf(raw_ptr: *const c_char) -> PathBuf {
+    let c_str = unsafe { CStr::from_ptr(raw_ptr) };
+    c_str.to_string_lossy().into_owned().into()
+}
+
+pub fn _pathbuf_array(raw_ptr: *mut *mut c_char) -> Vec<PathBuf> {
+    _convert_raw_ptr_array(raw_ptr, _pathbuf)
+}
+
+fn _convert_raw_ptr_array<F, R>(raw_ptr: *mut *mut c_char, convert: F) -> Vec<R>
+where
+    F: Fn(*const c_char) -> R,
+{
+    let mut ret_val: Vec<R> = vec![];
```

Comment:
Is there a reason that this couldn't just be `let mut ret_val = Vec::new();`?

Reply:
Probably not